### PR TITLE
flexible timesteps

### PR DESCRIPTION
**Pull request by ckingdon95**

### CODE REVIEW

**Comment by ckingdon95** [comment]
issue #54

File: src/clock.jl [hunk 1]
```diff
@@ -1,21 +1,82 @@
-type Clock
+type Timestep{Offset, Final}
```
Comment (davidanthoff) [inline src/clock.jl]: Should be an ``immutable``.

File: src/clock.jl [hunk 2]
```diff
@@ -1,21 +1,82 @@
...
+	function Clock(final::int)
+		ts = Timestep{1, final}(1)
+	end
+
+	function Clock(offset::int, final::int)
+		ts = Timestep{offset, final}(offset)
+	end
+end
+
+function getoffset(ts::Timestep)
```
Comment (davidanthoff) [inline src/clock.jl]: ``function foo{T}(x::Foo{T})``

File: src/clock.jl [hunk 3]
```diff
@@ -1,21 +1,82 @@
-type Clock
+type Timestep{Offset, Final}
 	t::Int
-	final_t::Int
+	final::Int
```
Comment (davidanthoff) [inline src/clock.jl]: Remove this field.

File: src/clock.jl [hunk 4]
```diff
@@ -1,21 +1,82 @@
...
+
+	function Timestep(i::Int)
+		ts = new{Offset, Final}()
+		ts.t = i - Offset + 1
+		ts.final = Final - Offset + 1
+		return ts
+	end
+
+	function Timestep()
```
Comment (davidanthoff) [inline src/clock.jl]: Remove this

File: src/clock.jl [hunk 5]
```diff
@@ -1,21 +1,82 @@
...
+		ts.final = Final - Offset + 1
+		return ts
+	end
+
+end
+
+type Clock
+	ts::Timestep
+
+	function Clock(final::int)
```
Comment (davidanthoff) [inline src/clock.jl]: Remove

File: src/mimi-core.jl [hunk 6]
```diff
@@ -197,6 +197,16 @@ function addcomponent(m::Model, t, name::Symbol=t.name.name; before=nothing,afte
     ComponentReference(m, name)
 end
 
+function addcomponent(m::Model, t, name::Symbol, offset::Int; before=nothing,after=nothing)
```
Comment (davidanthoff) [inline src/mimi-core.jl]: Make ``offset`` a keyword argument.

**Pull request merged**